documentation/specs/Notification.md

```diff
@@ -0,0 +1,567 @@
+    showNeutralAlert: (content) =>
+      queue.alert({ ...content, status: "neutral", type: "alert" }),
+    showSuccessAlert: (content) =>
+      queue.alert({ ...content, status: "success", type: "alert" }),
+    showWarningAlert: (content) =>
+      queue.alert({ ...content, status: "warning", type: "alert" }),
+    showErrorAlert: (content) =>
+      queue.alert({ ...content, status: "error", type: "alert" }),
+    showPromotionalToast: (content) =>
+      queue.toast({ ...content, status: "promotional", type: "toast" }),
+    showNeutralToast: (content) =>
+      queue.toast({ ...content, status: "neutral", type: "toast" }),
+    showSuccessToast: (content) =>
+      queue.toast({ ...content, status: "success", type: "toast" }),
+    showWarningToast: (content) =>
+      queue.toast({ ...content, status: "warning", type: "toast" }),
+    showErrorToast: (content) =>
+      queue.toast({ ...content, status: "error", type: "toast" }),
+  };
+}
+
+export type NotificationPositionOffset = {
+  top?: string;
+  right?: string;
+  bottom?: string;
+  left?: string;
+};
+
+export type NotificationProviderProps = {
+  children: ReactNode;
+  notificationPlacementOffset?: NotificationPositionOffset;
+};
+
+export function NotificationProvider(props: NotificationProviderProps) {
+  const { children, notificationPlacementOffset } = props;
+  let notification = useNotificationState();
+
+  return (
+    <NotificationContext.Provider value={notification}>
+      <NotificationContainer
+        notificationPlacementOffset={notificationPlacementOffset}
+      />
+      {children}
+    </NotificationContext.Provider>
+  );
+}
+
+export const useNotification = () => {
+  const notification = useContext(NotificationContext)!;
+  return { notification };
+};
+
+export type NotificationContainerProps = {
+  notificationPlacementOffset?: NotificationPositionOffset;
+};
+
+export function NotificationContainer(props: NotificationContainerProps) {
+  const { notificationPlacementOffset = null } = props;
+  const { notification } = useNotification();
+  const positionStyleProps = notificationPlacementOffset
+    ? {
+        top: notificationPlacementOffset?.top,
+        right: notificationPlacementOffset?.right,
+        bottom: notificationPlacementOffset?.bottom,
+        left: notificationPlacementOffset?.left,
+      }
+    : undefined;
+  // again, `visibleToasts` is an artifact of react-stately
+  return (
+    <>
+      {notification.visibleToasts.length > 0
+        ? ReactDOM.createPortal(
+            <div style={positionStyleProps}>
+              <NotificationRegion notification={notification} />{" "}
+            </div>,
+            document.body,
+          )
+        : null}
+    </>
+  );
+}
+```
+
+### Example Usage
+
+The examples below assume your app is consuming `EasyUIProvider`. This is important because `NotificationProvider` is rendered at the root of `EasyUIProvider`.
```

Comment:
**thought:** fwiw the Provider will be required for a lot of different functionality. we'll make it clear in the installation instructions that it's a required piece of integrating Easy UI.